File: client/Vpyqt/core/rank.py
import uuid
from core.config import getConfig, setConfig,request
import uuid
from core.LLMCaller import LLMCaller
from core.provider import deepseek
from PySide6.QtCore import QObject, Signal, Slot, QThread,QDate

class RankManager(QObject):
    def __init__(self) -> None:
        super().__init__()
    
    def getScores(self):
        email = getConfig()["USER"]["EMAIL"]
        if email=="":
            pass
        else:
            data = {
                "email":getConfig()["USER"]["EMAIL"]
            }
            res = request("get_scores",json=data)
            res = res.json()
            if res["code"]==200:
                return res["data"]
            else:
                return None

    def getRank(self):
        Scores = self.getScores()
        # print(Scores)
        rank = sorted(Scores, key=lambda x: x["score"], reverse=True)
        # print(rank)
        return rank

File: client/Vweb/js/core/rank.js
import { api } from './api.js';
import { state } from './state.js';

export async function getRankings() {
    const res = await api.getScores({ email: state.session.email });
    if (res.code !== 200) {
        throw new Error('获取排行榜失败: ' + res.msg);
    }
    const rankings = res.data; // 假设返回的数据结构是 { code: 200, scores: [...] }
    // 对数据进行格式化
    return rankings;
}